pkg/skaffold/jib/jib.go

```diff
@@ -17,32 +17,114 @@ limitations under the License.
 package jib
 
 import (
+	"encoding/json"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"sort"
+	"time"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 	"github.com/karrick/godirwalk"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
-func getDependencies(cmd *exec.Cmd) ([]string, error) {
-	stdout, err := util.RunCmdOut(cmd)
-	if err != nil {
-		return nil, err
+// filesTemplate contains fields parsed from Jib's JSON output
+type filesTemplate struct {
+	Build  []string
+	Inputs []string
+	Ignore []string
+}
+
+// filesLists contains cached build/input dependencies
+type filesLists struct {
+	WatchedBuildFiles []string
+	WatchedInputFiles []string
+	BuildFileTimes    map[string]time.Time
+}
+
+// watchedFiles maps from project name to watched files
+var watchedFiles = map[string]filesLists{}
```

Comment:
Learning go and the repo pattern.

Is it customary to have a global variable for this purpose? It may be fine here in terms of incorrect overwriting/sharing states between components, but I wonder if it isn't better to carry a state in `context.Context` (basically a new derived `Context` from the given one) which is probably well scoped and makes us free of worry of whether the code here can be executed concurrently. Just throwing an idea that came to my mind while trying to get familiar with the repo.

Reply:
It *seems* a possibility that local builds may build multiple artifacts in parallel in the future?

https://github.com/GoogleContainerTools/skaffold/blob/v0.24.0/pkg/skaffold/build/local/local.go#L45

Reply:
+1 that we'd like to avoid global vars unless they are absolutely required.
I would recommend moving this state into a struct and convert the methods to instance methods instead.

Reply:
Where would be a good place to create an instance of the struct?

Reply:
That's not trivial because we currently don't have a state per artifact

Reply:
In that case, should I just leave it for now and file an issue?

Reply:
I think you can use a `sync.Map`

Reply:
Updated to use `sync.Map`.